metadata/metadata.c

```diff
@@ -299,22 +318,21 @@ void _generate_and_set_disk_range_key(DiskRangeKey *key, ProtobufCBinaryData par
 
 PhysicalDiskRange *_initialize_next_physical_disk_range(Metadata *metadata,
         PhysicalDisk *disk,
-        Allocator *allocator,
         int disk_index)
 {
     PhysicalDiskRange *range = _new_physical_disk_range();
     _generate_and_set_disk_range_key(range->key, disk->key);
-    _initialize_disk_range_size(range->size, allocator->physical_disk_range_sector_count, disk_index);
+    _initialize_disk_range_size(range->size, metadata_disk_allocator->physical_disk_range_sector_count, disk_index);
     _add_physical_disk_range(metadata, range);
     _add_unallocated_physical_disk_range(disk, range);
 
     return range;
 }
 
-PhysicalDisk *create_physical_disk(Metadata *metadata, NVMfTransport *transport, uint64_t sector_count, uint64_t sector_size, Allocator *allocator)
+PhysicalDisk *create_physical_disk(Metadata *metadata, NVMfTransport *transport, uint64_t sector_count, uint64_t sector_size)
```

Comment:
The goal of hiding the allocator from the user would be already achieved by removing the allocator parameter from the create_physical_disk function only. I would still pass it as a parameter to all the underscore functions since it might come in handy once we do want to support different allocators for different virtual disks.